packages/@aws-cdk/aws-glue/lib/job-executable.ts

```diff
@@ -0,0 +1,415 @@
+   */
+  public static of(version: string): GlueVersion {
+    return new GlueVersion(version);
+  }
+
+  /**
+   * The name of this GlueVersion, as expected by Job resource.
+   */
+  public readonly name: string;
+
+  private constructor(name: string) {
+    this.name = name;
+  }
+}
+
+/**
+ * Runtime language of the Glue job
+ */
+export enum JobLanguage {
+  /**
+   * Scala
+   */
+  SCALA = 'scala',
+
+  /**
+   * Python
+   */
+  PYTHON = 'python',
+}
+
+/**
+ * Python version
+ */
+export enum PythonVersion {
+  /**
+   * Python 2 (the exact version depends on GlueVersion and JobCommand used)
+   */
+  TWO = '2',
+
+  /**
+   * Python 3 (the exact version depends on GlueVersion and JobCommand used)
+   */
+  THREE = '3',
+}
+
+/**
+ * The job type.
+ *
+ * If you need to use a JobType that doesn't exist as a static member, you
+ * can instantiate a `JobType` object, e.g: `JobType.of('other name')`.
+ */
+export class JobType {
+  /**
+   * Command for running a Glue ETL job.
+   */
+  public static readonly ETL = new JobType('glueetl');
+
+  /**
+   * Command for running a Glue streaming job.
+   */
+  public static readonly STREAMING = new JobType('gluestreaming');
+
+  /**
+   * Command for running a Glue python shell job.
+   */
+  public static readonly PYTHON_SHELL = new JobType('pythonshell');
+
+  /**
+   * Custom type name
+   * @param name type name
+   */
+  public static of(name: string): JobType {
+    return new JobType(name);
+  }
+
+  /**
+   * The name of this JobType, as expected by Job resource.
+   */
+  public readonly name: string;
+
+  private constructor(name: string) {
+    this.name = name;
+  }
+}
+
+interface SharedJobExecutableProps {
+  /**
+   * Glue version.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/release-notes.html
+   */
+  readonly glueVersion: GlueVersion;
+
+  /**
+   * Specifies the Amazon Simple Storage Service (Amazon S3) path to a script that executes a job.
+   */
+  readonly scriptLocation: string;
+
+  /**
+   * The Amazon S3 paths to additional files, such as configuration files that AWS Glue copies to the working directory of your script before executing it.
+   * Only individual files are supported, not a directory path.
+   *
+   * @default - no extra files and argument is not set
+   *
+   * @see `--extra-files` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly extraFiles?: string[];
+}
+
+interface SharedSparkJobExecutableProps extends SharedJobExecutableProps {
+  /**
+   * The Amazon S3 paths to additional Java .jar files that AWS Glue adds to the Java classpath before executing your script.
+   * Only individual files are supported, not a directory path.
+   *
+   * @default - no extra jars and argument is not set
+   *
+   * @see `--extra-jars` in https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly extraJars?: string[];
```

Comment:
also should be `Code`-like

Reply:
Resolving in favour of the comment on `scriptLocation`